docs/providers/aws/events/sns.md

```diff
@@ -168,3 +184,19 @@ resources:
       Properties:
         QueueName: myDLQ
 ```
+
+Alternatively, you can specify both `deadLetterTargetArn` and `deadLetterTargetUrl` using strings or any supported CloudFormation function like `Fn::ImportValue`
+
+```yml
+functions:
+  dispatcher:
+    handler: dispatcher.handler
+    events:
+      - sns:
+          topicName: dispatcher
+          redrivePolicy:
+            deadLetterTargetArn:
+              'Fn::ImportValue': MyShared-DLQArn
+            deadLetterTargetUrl:
+              'Fn::ImportValue': MyShared-DLQUrl
```

Comment:
Is there a case, where complete configuration must include a URL?

I wonder why do we allow to provide both ARN and URL? Is ARN not enough for some cases?

Reply:
If you pass an Arn with ImportValue or a Join for example(i had to use importvalue in one of my projects), then I haven't found a reliable way to generate the url from that Arn since it could use a different region/name than your current deployment.
Maybe I am missing some CloudFormation function that allows to do that?

Reply:
I think this naming convention could be a little confusing when trying to specify the arn with Fn::Join. 
I would prefer `deadLetterTargetImport` to be named something else where you can use any way of defining your arn and your url. What about `deadLetterTargetCustomDef`?
so then the arn and url fields inside can be specified with any cloudformation function like Join Import or whatever else

Reply:
> I think this naming convention could be a little confusing when trying to specify the arn with Fn::Join.

What would be the use case for user wanting to use `Fn::Join` when having options as above? I believe in such case user should just reference resource id via `deadLetterTargetRef: myDLQ` (?)

> to be named something else where you can use any way of defining your arn and your url. 

To avoid unnecessary complexity in handling that, and to make things more straightforward, I think it's better to lay out things more to the point as:
- If you want to reference via ARN, use `deadLetterTargetArn`
- If you want to reference by pointing an existing resource in same stack, use `deadLetterTargetRef`
- If you want to reference by pointing an output from other stack, use `deadLetterTargetImport`

Do you think it won't cover all cases?

Reply:
Well the Fn::Join case could be useful when generating the arn from different variables, but well I don't want to argue that much since most of the cases will be covered anyway. I'll updtae the PR asap